src/views/catalog/wizard/tabs/network/components/list/NetworkInterfaceActions.tsx

```diff
@@ -50,11 +52,18 @@ const NetworkInterfaceActions: FC<NetworkInterfaceActionsProps> = ({
 
   const onDelete = useCallback(() => {
     const updatedVM = produceVMNetworks(vm, (draftVM) => {
-      draftVM.spec.template.spec.networks = draftVM.spec.template.spec.networks.filter(
+      const vmInterfaces = getInterfaces(draftVM);
+      if (isEmpty(vmInterfaces)) {
```

Comment:
I don't see any distinction between IT and Template. Shouldn't there be one? as this action of network effect all VM's

Reply:
In Template mode you will not end up in this case. 
In template mode the default interface is specified in the yaml. 
You'll end up here only when you are trying to remove the default interface but the VM has nothing in the array because that interface is implicitly declared . Hope this make sense